packages/valory/skills/transaction_settlement_abci/behaviours.py

```diff
@@ -253,19 +253,19 @@ def _get_tx_data(
         }
 
         # Set hash, nonce and tip.
-        self.params.tx_hash = cast(str, tx_data["tx_digest"])
-        if nonce == self.params.nonce:
+        self.params.mutable_params.tx_hash = cast(str, tx_data["tx_digest"])
```

Comment:
It's not ideal to combine mutable and immutable data in the base params. I separated them now. Makes it more explicit what is configuration.

Reply:
why are we storing these in params, shared state makes more sense, no? Also makes the interface clearer IMO, params are always the same, you store state in SharedState.

Reply:
I agree somewhat. Although practically it is messier here to add shared state attributes.